rust/common/error/macros.rs

```diff
@@ -108,4 +131,65 @@ macro_rules! error_messages {
             return format!($body, _0, _1, _2, _3)
         }
     };
+
+    (@payload $self:ident, $format:ident, $error_name:ident) => {
+        if let Self::$error_name() = &$self {
+            return $format.field("payload", &())
+        }
+    };
```

Comment:
I wonder if this can be merged with `@format` into a more generic helper that is invoked as e.g. 
```
error_messages!(@dispatch self, $error_name $(, $inner)*, debug_struct.field("payload", &payload));)*
```

Reply:
Might be feasible using an immediately invoked lambda in the implementation.

Reply:
Now that writing the payload field has been inlined, `@payload` and `@format` seem too different to be reconciled. `@format` returns a value, whereas `@payload` now modifies the `&mut DebugStruct` passed to it.

Reply:
We've agreed in-person that there is no easy way to unify `@format` with `@payload`. The fundamental issue was in dealing with the unknown number of values wrapped by the enum instance.